src/query_engine/query_element/RemoteIterator.h

```diff
@@ -0,0 +1,77 @@
+#ifndef _QUERY_ELEMENT_REMOTEITERATOR_H
+#define _QUERY_ELEMENT_REMOTEITERATOR_H
+
+#include "QueryElement.h"
+
+using namespace std;
+
+namespace query_element {
+
+/**
+ * A special case of QueryElement because RemoteIterator is not actually an element of the
+ * query tree itself but rather an utility class used to remotely connect to the sink of a query
+ * tree (RemoteSink).
+ *
+ * Bacisally, the goal of this class is to allow a caller to request a query execution remotely
```

Comment:
```suggestion
 * A special case of QueryElement because RemoteIterator is not actually an element of the
 * query tree itself but rather a utility class used to remotely connect to the sink of a query
 * tree (RemoteSink).
 *
 * Basically, the goal of this class is to allow a caller to request a query execution remotely
```